src/electron/windows/sign_windows_executable.mjs

```diff
@@ -0,0 +1,204 @@
+}
+
+function assert(condition, msg) {
+  if (!condition) {
+    throw new Error(msg);
+  }
+}
+
+async function assertFileExists(file, msg) {
+  try {
+    await access(file, constants.R_OK | constants.W_OK);
+  } catch (err) {
+    throw new Error(format(msg, file), {cause: err});
+  }
+}
+
+/**
+ * Get the required option value from either cliValue or environment variable.
+ * @param {object} options the CLI options object.
+ * @param {string} argName the CLI argument name.
+ * @param {string} envName the environment variable name for this option.
+ * @param {boolean} required indicates whether this option is required.
+ * @returns {string} the value of the option
+ */
+function getOptionValue(options, argName, envName, required) {
+  const cliValue = options ? options[argName] : null;
+  const v = cliValue ?? process.env[envName];
+  if (required) {
+    assert(!!v, `either --${argName} or ${envName} is required`);
+  }
+  return v;
+}
+
+function appendPfxJsignArgs(args, options) {
+  // self-signed development certificate
+  args.push('--storetype', 'PKCS12');
+
+  const pfxCert = getOptionValue(options, 'pfx', 'WINDOWS_SIGNING_PFX_CERT', true);
+  args.push('--keystore', pfxCert);
+}
+
+function appendDigicertUsbJsignArgs(args, options) {
+  // extended validation certificate stored in USB drive
+  args.push('--storetype', 'PKCS11');
+
+  const subject = getOptionValue(options, 'subject', 'WINDOWS_SIGNING_EV_CERT_SUBJECT', false);
+  if (subject) {
+    args.push('--alias', subject);
+  }
+
+  var eTokenCfg;
+  switch (process.platform) {
+    case 'win32':
+      eTokenCfg = resolve(currentDirname(), 'digicert-usb-config', 'eToken-windows.cfg');
+      break;
+    case 'darwin':
+      eTokenCfg = resolve(currentDirname(), 'digicert-usb-config', 'eToken-macos.cfg');
+      break;
+    default:
+      throw new Error(`we do not support ev signing on ${process.platform}`);
+  }
+  args.push('--keystore', eTokenCfg);
+}
+
+/**
+ * Run jsign.jar according to the corresponding options targeting fileToSign.
+ * @param {string} fileToSign The path string of a file to be signed.
+ * @param {string[]} options The options to be passed to jsign. see https://ebourg.github.io/jsign/
+ * @returns {Promise<number>} A promise containing the exit code of jsign.
+ */
+function jsign(fileToSign, options) {
+  if (!options) {
+    throw new Error('options are required by jsign');
+  }
+  if (!fileToSign) {
+    throw new Error('fileToSign is required by jsign');
+  }
+
+  const jSignJarPath = resolve(outlineDirname(), 'third_party', 'jsign', 'jsign-4.0.jar');
+  const jsignProc = spawn(
+      'java',
+      [
+        '-jar', jSignJarPath,
+        ...options,
+        fileToSign,
+      ],
+      {
+        stdio: 'inherit',
+      });
+  jsignProc.on('error', err => console.error(`failed to start java due to "${err}", ` +
+      'please make sure you have java installed'));
+  return new Promise(resolve => jsignProc.on('exit', resolve));
```

Comment:
Perhaps call reject on error?

Reply:
updated, and will catch the error in the caller.